server.js

```diff
@@ -59,6 +59,55 @@ function startWorker(id) {
 
   const express = require('express');
   const Promise = require('bluebird');
+  const Profiler = require('./lib/tools/profiler');
+  const Uploader = require('./lib/tools/uploader');
+
+  /**
+   * Profiler setup
+   */
+
+  if (process.env.V8_PROFILER_ENABLED &&
+      process.env.AWS_ACCESS_KEY_ID &&
+      process.env.AWS_SECRET_ACCESS_KEY) {
+    logger.info('HEAP SNAPSHOT PROFILER IS ENABLED.');
+
+    const profiler = new Profiler(null, `worker #${id}`, 'heapsnapshot');
+    const uploader = new Uploader({
+      params: {
+        Bucket: 'v8-profiler.dosomething.org',
+        ContentType: 'binary/octet-stream',
+        ACL: 'public-read',
+      },
+    });
+
+    /**
+     * The profiler has fired the snapshotCreated event, now we can process the snapshot
+     */
+    profiler.on('snapshotCreated', (snapshot) => {
+      const fileName = snapshot.getHeader().title;
+      const uploadStream = uploader.getUploadStream(fileName);
+
+      /**
+       * This is the multipart stream that will upload the snapshot to S3
+       */
+      uploadStream
+        .on('error', (error) => {
+          logger.info(`Upload Stream error: ${error}`);
+          snapshot.delete();
+        })
+        .on('uploaded', () => snapshot.delete());
```

Comment:
It'd be useful to get a log message upon upload success, maybe just `logger.debug`. I don't have access to the Gambit S3 yet, so can't verify whether anything's been uploaded by looking at the logs. Not sure if it makes sense to `debug` here or in `snapshot.export().pipe` (or both)...